servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyPipelineSslUtils.java

```diff
@@ -60,7 +60,21 @@ public static SSLSession extractSslSessionAndReport(ChannelPipeline pipeline,
                                                         SslHandshakeCompletionEvent sslEvent,
                                                         Consumer<Throwable> failureConsumer,
                                                         boolean shouldReport) {
-        final SecurityHandshakeObserver observer = shouldReport ? handshakeObserver(pipeline) : null;
+        final SecurityHandshakeObserver observer;
+        if (shouldReport) {
+            try {
+                observer = handshakeObserver(pipeline);
+            } catch (Exception e) {
+                if (!sslEvent.isSuccess()) {
+                    e.addSuppressed(sslEvent.cause());
+                }
+                deliverFailureCause(failureConsumer, e, null);
```

Comment:
We should avoid failing the normal operations due to issues with observability. Doing this make the observability failures non-recoverable which isn't desirable as opposed to continuing with degraded observability.

Reply:
There is nothing that users can do to misconfigure and hit this error. If they pass a custom transport observer, it won't happen. Also, users don't have ways to modify the pipeline.

This is only for us, as developers, to make sure that new protocols that we add do not have misconfiguration around observability. Any test that uses SSL for a new protocol will fail if `shouldReport` is true, but pipeline does not have a handler. It helps to avoid writing an additional test to verify that new protocol reports handshake status to `SecurityHandshakeObserver`.

I don't have a strong opinion here, can remove it if it's not necessary.

Reply:
known errors are avoidable, unknown errors are the problem with such things. Being pessimistic is better, so will suggest removing.